src/gsCore/gsPolyField.h

```diff
@@ -0,0 +1,193 @@
+        m_fields.push_back(memory::make_shared_not_owned(&f));
+        m_names.push_back(name);
+        m_isparams.push_back(isparam);
+    }
+
+    /// \brief Number of patches in the geometry
+    index_t nPatches() const { return m_patches->nPieces(); }
+
+    /// \brief Number of fields in the container
+    size_t nFields() const { return m_fields.size(); }
+
+    /// \brief Returns the geometry patch number \a p.
+    const gsGeometry<T> & patch(index_t p) const
+    {
+        GISMO_ASSERT(p >= 0 && p < nPatches(),
+                     "Invalid patch index " + util::to_string(p) +
+                     ". Number of patches is " + util::to_string(nPatches()) +".\n");
+        return static_cast<const gsGeometry<T> &>(m_patches->piece(p));
+    }
+
+    /// \brief Returns the function corresponding to the geometry patch \a p from the field \a i.
+    const gsFunction<T> & function(index_t p, size_t i) const
+    {
+        GISMO_ASSERT(p >= 0 && p < nPatches(),
+                     "Invalid patch index " + util::to_string(p) +
+                     ". Number of patches is " + util::to_string(nPatches()) +".\n");
+        GISMO_ASSERT(i >= 0 && i < nFields(),
+                     "Invalid field index " + util::to_string(i) +
+                     ". Number of fields is " + util::to_string(nFields()) +".\n");
+        return static_cast<const gsFunction<T> &>(m_fields[i]->piece(p));
+    }
+
+    const gsGeometry<T> & igaFunction(index_t p, size_t i) const
+    {
+        GISMO_ASSERT(p >= 0 && p < nPatches(),
+                     "Invalid patch index " + util::to_string(p) +
+                     ". Number of patches is " + util::to_string(nPatches()) +".\n");
+        GISMO_ASSERT(i >= 0 && i < nFields(),
+                     "Invalid field index " + util::to_string(i) +
+                     ". Number of fields is " + util::to_string(nFields()) +".\n");
+        GISMO_ASSERT(m_isparams[i],"Field " + util::to_string(i) + " is not parametric.\n");
+        return static_cast<const gsGeometry<T> &>(m_fields[i]->piece(p));
+    }
+
+    /// \brief Returns the name of the field number \a i.
+    const std::string & name(size_t i) const
+    {
+        GISMO_ASSERT(i >= 0 && i < nFields(),
+                     "Invalid field index " + util::to_string(i) +
+                     ". Number of fields is " + util::to_string(nFields()) +".\n");
+        return m_names[i];
+    }
+
+    /// \brief Returns true if the field number \a i is parametric.
+    bool isParametric(size_t i) const
+    {
+        GISMO_ASSERT(i >= 0 && i < nFields(),
+                     "Invalid field index " + util::to_string(i) +
+                     ". Number of fields is " + util::to_string(nFields()) +".\n");
+        return m_isparams[i];
+    }
+
+    /// \brief Returns true if the field number \a i is a BSpline function.
+    bool isParametrized(size_t i) const // bad function.
```

Comment:
// bad function.
?

Reply:
It breaks polymorphism by explicitly asking whether a gsFunction object is a gsGeometry. This is used in one place only - in plotting a mesh to Paraview, same as for the original gsField.